### PR TITLE
Pickup requests cancel

### PR DESCRIPTION
**Pull request by lauriek334**
Added timestamp string to the cancel_time field when a hold_request is cancelled. Previously, only the date string was being passed. This is initiated if the user clicks on the "Cancel Holds" email link or from the arborcat holdsblock.
This has been tested on pinkeye and timezone settings have been verified as being correct.  Currently this is installed on pinkeye and is available for further testing if needed.

### CODE REVIEW

**Comment by taleon** [comment]
Not sure what that php cache file is, but guessing unintentionally committed. I'd probably change the function name instances of patronId to patron_id.

**Comment by lauriek334** [comment]
Removed cache file that was accidentally added to the commit.
Changed method names to have patron_id

**Comment by taleon** [comment]
Only comments here are a couple instances of var names of patronId and not patron_id, and try/catches which return no useful error message in case there is an API issue.

**Comment by lauriek334** [comment]
Removed commented out code with debugging code

File: arborcat.module [hunk 1]
```diff
@@ -347,6 +347,7 @@ function arborcat_create_pickup_request_record($pickup_request_type, $custom_id,
 
       return $id;
     } catch (Exception $e) {
+      drupal_set_message('An error occurred saving this pickup request. Please try again');
```
Comment (ejk) [inline arborcat.module:350]: You can add a second argument of 'error' to drupal_set_message() to make the message display with error styling. (and I know drupal_set_message() will be deprecated in Drupal 9)

File: arborcat.module [hunk 2]
```diff
@@ -355,55 +356,53 @@ function arborcat_create_pickup_request_record($pickup_request_type, $custom_id,
...
+    $return_val = $patron_id;
   }
+
+  return $return_val;
 }
 
-  
 function arborcat_custom_pickup_request($pickup_request_type, $custom_pickup_request_id) {
-  $result_message = '';
+  $resultMessage = '';
```
Comment (ejk) [inline arborcat.module:405]: Looks like this variable name got switched from underscores to camelCase

File: src/Form/LockerRequestForm.php [hunk 3]
```diff
@@ -219,7 +219,7 @@ function arborcat_mail($key, $email_to, $patron, $code, $holds) {
 			foreach($holds as $title){
 				$email_message = $email_message.$title['Title']."\r\n";
 			}
-			$mailManager = \Drupal::service('plugin.manager.mail');
+			$mail_manager = \Drupal::service('plugin.manager.mail');
```
Comment (ejk) [inline src/Form/LockerRequestForm.php:222]: Looks like this is the only reference to this variable. Does this line even need to be here?

File: src/Form/UserPickupRequestForm.php [hunk 4]
```diff
@@ -395,7 +402,7 @@ public function arborcat_mail($key, $email_to, $patron, $code, $holds) {
       foreach ($holds as $title) {
         $email_message = $email_message.$title['Title']."\r\n";
       }
-      $mailManager = \Drupal::service('plugin.manager.mail');
+      $mail_manager = \Drupal::service('plugin.manager.mail');
```
Comment (ejk) [inline src/Form/UserPickupRequestForm.php:405]: Again, looks like we are instantiating the drupal mail service but then just using the PHP mail() function on the next line. Can probably just delete this line.

**Review by ejk — COMMENTED** [review]
Looks good, pending pedantic in-line comments

**Comment by lauriek334** [comment]
Addressed your comments Eric - all except 1.
 - added the 'error' as a second parameter to drupal_set_message.  
- removed calls to Drupal MailManager.
- fixed resultMessage camelcase instance
- Will do a separate edit of all drupal_set_message calls in the arborcat module and replace them with the following in order to be ready for Drupal 9 (https://www.palantir.net/updating-deprecated-drupalsetmessage-calls-drupal-8):
\Drupal::messenger()->addStatus()
\Drupal::messenger()->AddError()
\Drupal::messenger()->AddWarning()

EJK Not sure what you are referring to - Pending pedantic in-line comments???

**Comment by ejk** [comment]
> EJK Not sure what you are referring to - Pending pedantic in-line comments???

My apologies, that was referring to MY pedantic comments on the Pull Request review. I sometimes feel I'm being picky so I was trying to be a little self-deprecating. 😄

**Comment by lauriek334** [comment]
Phew - happy to get all comments (pedantic or not).

**Comment by taleon** [comment]
So these seems good from limited testing and looking over it. We should probably get this merged into prodtest and make sure things will be good on the code cleanup branch. Given the variable changes everywhere, it would be good to make sure we test both these branches for all aspects of pickups again just to be safe.

**Comment by lauriek334** [comment]
I’ll merge this pull request into prodtest and run through a set of tests on pinkeye to ensure the following operations are working:
- creating a pickup request
- viewing pickup requests in accounts page
- cancelling a pickup request from accounts page panel
- running pickuphelper to see eligible holds for a patron
- running pickuphelper to see cancel a pickup request for a patron
- creating a printing service pickup request
- creating a shelf service pickup request
- creating a summer game prize pickup request (I may need help setting that up Eric)

I’ll also perform the same set of steps on the code cleanup branch and will report back.


Laurie Kirchmeier
lauriek334@gmail.com

**Comment by lauriek334** [comment]
Merged this branch into codeCleanup branch and installed on pinkeye for testing.
I have successfully run the following tests:
- creating a pickup request
- viewing pickup requests in accounts page
- cancelling a pickup request from accounts page panel
- running pickuphelper to see eligible holds for a patron
- running pickuphelper to cancel a pickup request for a patron
- creating a printing service pickup request
- creating a shelf service pickup request
- cancelling a printing service request from the user accounts page panel
- cancelling a shelf service request using PickupHelper

I have not run any tests of SG prize pickups - Eric could you test the following:
-     creating a summer game prize pickup request 
-     test cancelling a SG prize pickup request

**Pull request merged**